yetanotherco/aligned_layer
# refactor(aggregator): Aggregator retries

# Adds retry functions to the Aggregator

## Description

Add retries to rpc calls and connections within the Aggregator:
- Refactors `RetryWithData` to add a panic catch and use generic types.
- Creates wrapper functions for code that await fallible connections using retry functions.
- Adds tests using Anvil as a local testnet that check the returning error case for a call that successfully connects and executes (but does return the correct value), then kills anvil and asserts the failing transient error is returned.

### Notes:
- Exhaustively finding and distinguishing all Permanent and Transient errors still needs to be done. Issue: https://github.com/yetanotherco/aligned_layer/issues/1390
- Test for `InitializeNewTaskRetryable` and `ProcessNewSignatureRetryable` and commented out. This is due to an issue with the eigen-sdk where it panics if it loses connection. There is an issue here: https://github.com/yetanotherco/aligned_layer/issues/1361
- The unit tests for the retries require coordination are somewhat flaky and could be approved in a separate pr by separating the retryable logic into a separate function and testing it independent of retry library function: https://github.com/yetanotherco/aligned_layer/pull/1392

### To Test

- Run a local network and verify things work
- Run the unit test `cd core && go test -v`

## Type of change


## Checklist
